packages/cyberstorm/src/components/CustomSelect.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { ReactNode } from "react";
+import styles from "./componentStyles/CustomSelect.module.css";
+import { Button } from "./Button";
+import { CustomSelectItem, SelectItemProps } from "./CustomSelectItem";
+import { faChevronDown } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import * as Select from "@radix-ui/react-select";
+import { SelectItem } from "@radix-ui/react-select";
+
+export type SelectOption = {
+  value: string;
+  content: typeof CustomSelectItem;
+  displayValue?: string;
+};
+
+type SelectProps = {
+  colorScheme?: "default" | "defaultDark" | "primary";
+  defaultOpen?: boolean;
+  icon?: ReactNode;
+  options?: SelectOption[];
+  setValue: CallableFunction;
+  value: string;
+};
+
+export const CustomSelect: React.FC<SelectProps> = (props) => {
+  const { colorScheme, defaultOpen, icon, options, setValue, value } = props;
+
+  const selectItemElements = options
+    ? mapSelectData(options, colorScheme)
+    : null;
+
+  return (
+    <div className={styles.root}>
+      <Select.Root
+        defaultOpen={defaultOpen}
+        value={value}
+        onValueChange={(newValue) =>
+          setValue !== undefined ? setValue(newValue) : null
```

Comment:
Per the current typing, `setValue` can't be undefined, so `onValueChange={setValue}` should probably suffice.

Reply:
👍